lib/segment/src/id_tracker/mutable_id_tracker.rs

```diff
@@ -382,10 +406,15 @@ fn load_mappings(mappings_path: &Path) -> OperationResult<PointMappings> {
 /// iterator will not produce any more items.
 fn read_mappings_iter<R: Read>(
     mut reader: R,
-) -> impl Iterator<Item = OperationResult<MappingChange>> {
+) -> impl Iterator<Item = OperationResult<(MappingChange, u64)>> {
+    let mut position = 0;
+
     // Keep reading until end of file or error
     std::iter::from_fn(move || match read_entry(&mut reader) {
-        Ok(entry) => Some(Ok(entry)),
+        Ok((entry, read_bytes)) => {
+            position += read_bytes;
+            Some(Ok((entry, position)))
+        }
         // Done reading if end of file is reached
         Err(err) if err.kind() == io::ErrorKind::UnexpectedEof => None,
```

Comment:
Why not do something like this:

```rust
let mut position = reader.stream_position()?;

std::iter::from_fn(move || match read_entry(&mut reader) {
	Ok((entry, read_bytes)) => {
		position += read_bytes;
		Some(Ok(entry))
	}
	// Done reading if end of file is reached
	Err(err) if err.kind() == io::ErrorKind::UnexpectedEof => {
		reader.seek(SeekFrom::Start(position))?;
		None
	}
	Err(err) => Some(Err(err.into())),
}
```

It will require a few more syscalls, but one extra when starting reading a file and a few extras when handling an error should be acceptable, IMO.

Reply:
Fine suggestion. I like that it keeps the return type a bit more clean. Thanks.

Reply:
Not necessary, I can simply check if we've exhausted the reader or not.

Reply:
@ffuugoo done in <https://github.com/qdrant/qdrant/pull/6173/commits/40c187d1ce2710eaa0b0d4b5e3cc7adf225c79c1>